packages/libs/error-stack/src/hook/mod.rs

```diff
@@ -181,4 +183,50 @@ impl Report<()> {
 
         closure(&hook)
     }
+
+    #[cfg(all(any(feature = "std", feature = "hooks"), feature = "serde"))]
+    pub fn install_serde_hook<T: serde::Serialize + Send + Sync + 'static>() {
+        crate::serde::install_builtin_hooks();
+
+        #[cfg(feature = "std")]
+        let mut lock = SERDE_HOOK.write().expect("should not be poisoned");
+
+        // The spin RwLock cannot panic
+        #[cfg(all(not(feature = "std"), feature = "hooks"))]
+        let mut lock = SERDE_HOOK.write();
+
+        lock.insert_static::<T>();
+    }
+
+    #[cfg(all(any(feature = "std", feature = "hooks"), feature = "serde"))]
+    pub fn install_custom_serde_hook<F, T>(closure: F)
```

Comment:
Is it possible to use `impl Trait` syntax for `F`, so we can call `install_custom_serde_hook::<MyType>(...)`?